docs/admin/monitoring.rst

```diff
@@ -168,22 +168,43 @@ NodeStatus can be accessed using the JMX MBean object name
 NodeInfo MBean
 --------------
 
-The ``NodeInfo`` JMX MBean exposes information about the current node;
+The ``NodeInfo`` MXBean exposes information of a ``CrateDB`` node.
 
-NodeInfo can be accessed using the JMX MBean object name
-``io.crate.monitoring:type=NodeInfo`` and the following attributes:
-
- - ``ClusterStateVersion``
-
-   Provides the version of the current applied cluster state
-
- - ``NodeId``
+It can be accessed using the ``io.crate.monitoring:type=NodeInfo`` object
+name and has following attributes:
 
-   Provides the unique identifier of the node in the cluster
++------------------+------------------------------------------------------+
+| Name             | Description                                          |
++==================+======================================================+
+| ``id``           | The id of the node.                                  |
++------------------+------------------------------------------------------+
+| ``name``         | The name of the node.                                |
++------------------+------------------------------------------------------+
+| ``role``         | The role of the node.                                |
++------------------+------------------------------------------------------+
+| ``primaries``    | The number of primary shards located on the node.    |
++------------------+------------------------------------------------------+
+| ``replicas``     | The number of replica shards located on the node.    |
++------------------+------------------------------------------------------+
+| ``unassigned``   | The number of unassigned shards. If the role of the  |
+|                  | node is ``master`` this will show the total number   |
+|                  | of unassigned shards in the cluster, otherwise 0.    |
++------------------+------------------------------------------------------+
+| ``shards``       | Information about the shards located on the node.    |
++------------------+------------------------------------------------------+
 
- - ``NodeName``
+``shards`` returns a set of `CompositeData`_ objects containing detailed
+information of each shard located on the node with the following attributes:
 
-   Provides the human friendly name of the node
++------------------+------------------------------------------------------+
+| Name             | Description                                          |
++==================+======================================================+
+| ``id``           | The id of the shard.                                 |
++------------------+------------------------------------------------------+
+| ``size``         | The size in bytes of the shard.                      |
++------------------+------------------------------------------------------+
+| ``status``       | The status of the shard.                             |
```

Comment:
What kind of states are there?

In which format are we going to expose this to prometheus in the jmx exporter? Is the state going to be a propery that can be filtered on?

Reply:
My idea was to expose the shard related information in the following way to prometheus:

```
crate_node(name=shards_info, property=number_of_primary_shards) 1
crate_node(name=shards_info, property=number_of_replicas) 2
crate_node(name=shards_info, property=number_of_shards) 3
crate_node(name=shards_info, property=number_of_unassigned_shards) 0

crate_shards(id=1, property=size) 300
crate_shards(state=started) 1 
crate_shards(state=recovering) 1 
```

Reply:
Sounds good :+1: